File: OpenTokFs.ResponseTypes/OpenTokProjectDetails.cs
using System;

namespace OpenTokFs.ResponseTypes {
    public class OpenTokProjectDetails {
        public int Id { get; set; } = 0;
        public string Secret { get; set; } = "";
        public string Status { get; set; } = "";
        public string Name { get; set; } = "";
        public string Environment { get; set; } = "";
        public long CreatedAt { get; set; } = 0L;

        public DateTimeOffset GetCreationTime() => DateTimeOffset.FromUnixTimeMilliseconds(CreatedAt);

        public override string ToString() => $"{Id} ({Name})";
    }
}

File: OpenTokFs/ResponseTypes/OpenTokProjectDetails.fs
namespace OpenTokFs.ResponseTypes

open System
open OpenTokFs.Credentials

type OpenTokProjectDetails = {
    id: int
    secret: string
    status: string
    name: string
    environment: string
    createdAt: int64
} with
    interface IProjectCredentials with
        member this.ApiKey = this.id
        member this.ApiSecret = this.secret
    member this.GetCreationTime () = DateTimeOffset.FromUnixTimeMilliseconds this.createdAt
    override this.ToString () = $"{this.id} ({this.name})"